### PR TITLE
Reworks CNS Implant + Emagged Upgraded Cybernetic Heart

### PR DESCRIPTION
**Pull request by ItsMarmite**
## What Does This PR Do

- Changes all instances of anti_stun to anti_stam in the code in relation to the CNS implant. 

- Reworks the CNS implant to protect against large bursts of stamina damage. 

- Reworks the emagged upgraded cybernetic heart to cause the user to regenerate stamina passively. 

- Removes emagged upgraded cybernetic heart's ability to negate drowsiness and sleeping. 

- CNS will now cause a 'weaken' instead of a 'stun' to give a better visual effect of the weakness if the user is subject to an EMP. Duration has not been changed.

## Why It's Good For The Game
With the move towards stamina based combat, CNS and Emagged Heart have been left behind, not really fitting in the current direction combat is moving. As a result, this PR reworks both to fit this new direction.

## Changelog
:cl:
tweak: Reworked CNS Implant and Emagged Upgraded Cybernetic Heart to fit within new direction of stamina combat.
/:cl:

### CODE REVIEW

**Comment by SpringSkipper** [comment]
DO NOT nerf emagged cyberheart like that. It's already extremely niche and makes ions one-shot you, basically turning you into an IPC traitor. Disablers causing heart attacks ruins it entirely.

**Comment by Inadvizable** [comment]
> DO NOT nerf emagged cyberheart like that. It's already extremely niche and makes ions one-shot you, basically turning you into an IPC traitor. Disablers causing heart attacks ruins it entirely.

This is a buff, it makes the heart insanely good at removing stamina. We're talking 20 stamina instead of 2 here, an entire order of magnitude better.

**Comment by SpringSkipper** [comment]
I'm looking at the "people can hear heart" and "heart attack from disablers" nerfs in particular.

Keep in mind that to get one you need to
-research until you can get a heart, then print one
-emag your heart
-get someone else to put the heart in

and then you already instantly die to EMP and sometimes stun batons and door zaps. The moment you go down from disabler fire they're going to start bapping you with batons anyways. Why is that necessary?

and then you get outed to anyone standing next to you?

**Comment by Qwertytoforty** [comment]
> I'm looking at the "people can hear heart" and "heart attack from disablers" nerfs in particular.
> 
> Keep in mind that to get one you need to -research until you can get a heart, then print one -emag your heart -get someone else to put the heart in
> 
> and then you already instantly die to EMP and sometimes stun batons and door zaps. The moment you go down from disabler fire they're going to start bapping you with batons anyways. Why is that necessary?
> 
> and then you get outed to anyone standing next to you?

Noise was removed. PR description was not updated.

**Comment by RagingLemons1** [comment]
as others have said, everything looks fine except the nerf of the cyber heart. It already has a way to kill you with ions, with the hassle of getting one installed, you don't need to and shouldn't give sec disablers a heart attack function. 

no problem with the heart noise though, if you have one of these it seems assumed that you're going loud.

**Comment by procdrone** [comment]
Agreeing with others.

considering that every single officer is going to have a disabler and will 99% use it on people they chase, that particular malus for emmaged heart does seem excessive.

**Comment by hal9000PR** [comment]
Every officer also has a baton, which currently does the same thing.
Every officer used to have a taser, which did the same thing.

File: code/modules/projectiles/projectile/beams.dm [hunk 1]
```diff
@@ -55,6 +55,11 @@
 	impact_effect_type = /obj/effect/temp_visual/impact_effect/blue_laser
 	light_color = LIGHT_COLOR_CYAN
 
+/obj/item/projectile/beam/disabler/on_hit(atom/target, blocked, hit_zone)
```
Comment (Fox-McCloud) [inline code/modules/projectiles/projectile/beams.dm:58]: This is insane. Don't do this.

Comment (ItsMarmite) [reply code/modules/projectiles/projectile/beams.dm:58]: If the head coders want it removed, I'll do so, but it was implemented for consistency with how tasers previously interacted with the emagged upgraded cybernetic heart.

Remove it, or keep it?

**Review by hal9000PR — CHANGES_REQUESTED** [review]

File: code/modules/surgery/organs/augments_internal.dm [hunk 2]
```diff
@@ -29,10 +29,10 @@
 /obj/item/organ/internal/cyberimp/brain/emp_act(severity)
 	if(!owner || emp_proof)
 		return
-	var/stun_amount = (5 + (severity-1 ? 0 : 5)) STATUS_EFFECT_CONSTANT
-	owner.Stun(stun_amount)
+	var/weaken_time = (5 + (severity-1 ? 0 : 5)) STATUS_EFFECT_CONSTANT
```
Comment (Sirryan2002) [inline code/modules/surgery/organs/augments_internal.dm:32]:
```suggestion
	var/weaken_time = (5 + (severity - 1 ? 0 : 5)) STATUS_EFFECT_CONSTANT
```

File: code/modules/surgery/organs/augments_internal.dm [hunk 3]
```diff
@@ -120,42 +120,42 @@
...
-	desc = "This implant will automatically give you back control over your central nervous system, reducing downtime when stunned. Incompatible with the Neural Jumpstarter."
+	desc = "This implant will automatically give you back control over your central nervous system, reducing downtime when fatigued. Incompatible with the Neural Jumpstarter."
 	implant_color = "#FFFF00"
 	slot = "brain_antistun"
 	origin_tech = "materials=5;programming=4;biotech=5"
-	var/stun_max_amount = 4 SECONDS
-
-/obj/item/organ/internal/cyberimp/brain/anti_stun/hardened
+	var/last_stamina_damage = 0
+	var/max_stamina_increment = 40
```
Comment (Sirryan2002) [inline code/modules/surgery/organs/augments_internal.dm:130]:
```suggestion
	var/max_stamina_increment = 40

```

File: code/modules/surgery/organs/augments_internal.dm [hunk 4]
```diff
@@ -120,42 +120,43 @@
...
-/obj/item/organ/internal/cyberimp/brain/anti_stun/hardened
+/obj/item/organ/internal/cyberimp/brain/anti_stam/hardened
 	name = "Hardened CNS Rebooter implant"
 	emp_proof = TRUE
 
-/obj/item/organ/internal/cyberimp/brain/anti_stun/hardened/Initialize(mapload)
+/obj/item/organ/internal/cyberimp/brain/anti_stam/hardened/Initialize(mapload)
 	. = ..()
-	desc += " The implant has been hardened. It is invulnerable to EMPs."
+	desc += "This implant will automatically give you back control over your central nervous system, reducing downtime when fatigued. Incompatible with the Neural Jumpstarter. The implant has been hardened. It is invulnerable to EMPs."
```
Comment (hal9000PR) [inline code/modules/surgery/organs/augments_internal.dm:138]: this is not what I meant... this is
```
/obj/item/organ/internal/cyberimp/brain/anti_stam/hardened
	name = "Hardened CNS Rebooter implant"
	desc = "This implant will automatically give you back control over your central nervous system, reducing downtime when fatigued. Incompatible with the Neural Jumpstarter. The implant has been hardened. It is invulnerable to EMPs."
	emp_proof = TRUE
```

then remove the `init` override.

**Review by SteelSlayer — CHANGES_REQUESTED** [review]
Please address Hal's review within 7 days.

**Review by hal9000PR — APPROVED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Pull request merged**